### PR TITLE
🎨 Allow to have multiple consumers for the same queue in 1 application and allow to optionally define queue name

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
Before:
- The RabbitMQ consumer was setup with a name unique to the application, but this prevents from having multiple ones in the same application
After:
- the ConsumerTag is returned when subscribing to a Exchange/Queue
- unsubscribing a consumer from a Queue requires the ConsumerTag,
- Fixes unsubscribing when the connection to the RabbitMQ service is already shutdown,
- Upgraded derived services
- After discussion with @matusdrobuliak66 adds option for non-exclusive queues to define a specific queue name

## Related issue/s
- required by https://github.com/ITISFoundation/osparc-simcore/pull/6736

## Dev-ops checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: packages/service-library/src/servicelib/rabbitmq/_client.py [hunk 1]
```diff
@@ -138,20 +146,22 @@ async def _get_channel(self) -> aio_pika.abc.AbstractChannel:
             channel.close_callbacks.add(self._channel_close_callback)
             return channel
 
-    async def _get_consumer_tag(self, exchange_name) -> str:
-        return f"{get_rabbitmq_client_unique_name(self.client_name)}_{exchange_name}"
+    async def _create_consumer_tag(self, exchange_name) -> ConsumerTag:
+        return ConsumerTag(
+            f"{get_rabbitmq_client_unique_name(self.client_name)}_{exchange_name}_{uuid4()}"
```
Comment (GitHK) [inline packages/service-library/src/servicelib/rabbitmq/_client.py:151]: Nice that this is unique.

File: packages/service-library/src/servicelib/rabbitmq/_client.py [hunk 2]
```diff
@@ -238,20 +248,19 @@ async def subscribe(
...
+            consumer_tag = await self._create_consumer_tag(exchange_name)
             await queue.consume(
                 partial(_on_message, message_handler, unexpected_error_max_attempts),
                 exclusive=exclusive_queue,
-                consumer_tag=_consumer_tag,
+                consumer_tag=consumer_tag,
             )
-            output: str = queue.name
-            return output
+            return queue.name, consumer_tag
```
Comment (GitHK) [inline packages/service-library/src/servicelib/rabbitmq/_client.py:257]: Very nice trick. This allows for the consumer_tag to be local to the service where it is used and be unique. While the queue name does not change.

**Review by GitHK — APPROVED** [review]
Nice solution. This was such a pain!

File: packages/service-library/src/servicelib/rabbitmq/_models.py [hunk 3]
```diff
@@ -11,6 +11,11 @@
 
 MessageHandler = Callable[[Any], Awaitable[bool]]
 
+ExchangeName: TypeAlias = str
```
Comment (pcrespov) [inline packages/service-library/src/servicelib/rabbitmq/_models.py:14]: MINOR: should we add some constraints to these names? i.e. all identifiers should have minimum number of letters and allowed characters. That is the purpose of all constraint identifiers like `IDStr` etc

**Review by pcrespov — APPROVED** [review]
thx

**Pull request merged**